content/post/provisioning-redis-cluster-in-k8s/index.md

```diff
@@ -0,0 +1,87 @@
+of the instances is needed to create the cluster. So each instance should know the domain name or IP address of the other redis instances. The IP address or 
+the domain name is also needed to check if the instance is up and running.
+
+To locate the domain names of the database instances, we utilize a tool called [redis-node-finder](https://github.com/kubedb/redis-node-finder).
+Following that, it pings and determines how many instances are active.The cluster can be created by an instance if there are three up and running redis instances.
+Because all redis instances are attempting to crate clusters using their init scripts, the instance should confirm that the cluster does not 
+already exist before creating one. Only one of the instances is able to successfully create the cluster; the others join it after they discover that 
+the cluster is created.
+
+Each replica can join the cluster as a master or a replica after the cluster has been created. The cluster's initial instances are by default the master instances.
+If the current pod is the 0th pod in the statefulset, it joins the cluster as master since each statefulset should have one master according to our architecture.
+The other instances of the statefulset join the cluster as replica of the master.
+
+![Figure: Redis Cluster Initialization](redis-cluster-script.png)
+
+## Failover scenario
+
+In the event that the pod restarts, the init script should also join the pod in the cluster. Due to the HA solution we provide, when a pod resumes, there are two possible outcomes.
+The instance needed to be joined as a replica in the event of failover completion or replica node restarts. The instance should join as master in other scenarios.
+
+To handle the failover scenario for each instance, we first do a `cluster meet` command with all the current pods. The node's role is then verified.
+It joins the cluster if it is a master node. If not, we must determine whether the replica needs to be in sync with the master.
+When synchronization is required, the command "cluster replicate" is used to replicate all the data from the master.
+
+
+## Conclusion
+
+To enable Redis clustering we have used bash scripting from init script of the pods. We have identified different scenarios and handled them accordingly. 
+After provisioning the lifecycle of the redis cluster is maintained by KubeDB Operator.
+Additionally,  day-2 operations like Horizontal or Vertical Scaling, TLS operations can be done using KubeDB. To learn more, 
+you can have a look at [KubeDB Docs](https://kubedb.com/kubernetes/databases/run-and-manage-redis-on-kubernetes/).
+
+
+## Support
+
+To speak with us, please leave a message on [our website](https://appscode.com/contact/).
+
+To join public discussions with the KubeDB community, join us in the [Kubernetes Slack team](https://kubernetes.slack.com/messages/C8149MREV/) channel `#kubedb`. To sign up, use our [Slack inviter](http://slack.kubernetes.io/).
```

Comment:
~To join public discussions with the KubeDB community, join us in the [Kubernetes Slack team](https://kubernetes.slack.com/messages/C8149MREV/) channel `#kubedb`. To sign up, use our [Slack inviter](http://slack.kubernetes.io/).~